hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -181,75 +200,96 @@ update Environment{party, signingKey, otherParties} ledger st ev = case (st, ev)
+  (previousState@InitialState{parameters, committed}, OnChainEvent (Observation OnCollectComTx{})) ->
     -- TODO: We would want to check whether this even matches our local state.
     -- For example, we do expect `null remainingParties` but what happens if
     -- it's untrue?
     let u0 = fold committed
      in nextState
-          (OpenState parameters $ CoordinatedHeadState u0 mempty (InitialSnapshot $ Snapshot 0 u0 mempty) NoSeenSnapshot)
+          ( OpenState
+              { parameters
+              , coordinatedHeadState = CoordinatedHeadState u0 mempty (InitialSnapshot $ Snapshot 0 u0 mempty) NoSeenSnapshot
+              , previousState
+              }
+          )
           [ClientEffect $ HeadIsOpen u0]
-  (InitialState _ _ committed, OnChainEvent OnAbortTx{}) ->
+  (InitialState{committed}, OnChainEvent (Observation OnAbortTx{})) ->
     nextState ReadyState [ClientEffect $ HeadIsAborted $ fold committed]
   --
-  (OpenState _parameters CoordinatedHeadState{confirmedSnapshot}, ClientEvent Close) ->
+  (OpenState{coordinatedHeadState = CoordinatedHeadState{confirmedSnapshot}}, ClientEvent Close) ->
     sameState
       [ OnChainEffect (CloseTx confirmedSnapshot)
       ]
   --
-  (OpenState _ CoordinatedHeadState{confirmedSnapshot}, ClientEvent GetUTxO) ->
+  (OpenState{coordinatedHeadState = CoordinatedHeadState{confirmedSnapshot}}, ClientEvent GetUTxO) ->
     sameState
       [ClientEffect . UTxO $ getField @"utxo" $ getSnapshot confirmedSnapshot]
   --
-  (OpenState _ CoordinatedHeadState{confirmedSnapshot = getSnapshot -> Snapshot{utxo}}, ClientEvent (NewTx tx)) ->
+  (OpenState{coordinatedHeadState = CoordinatedHeadState{confirmedSnapshot = getSnapshot -> Snapshot{utxo}}}, ClientEvent (NewTx tx)) ->
     sameState effects
    where
     effects =
       case canApply ledger utxo tx of
         Valid -> [ClientEffect $ TxValid tx, NetworkEffect $ ReqTx party tx]
         Invalid err -> [ClientEffect $ TxInvalid{utxo = utxo, transaction = tx, validationError = err}]
-  (OpenState parameters headState@CoordinatedHeadState{seenTxs, seenUTxO}, NetworkEvent (ReqTx _ tx)) ->
+  (OpenState{parameters, coordinatedHeadState = headState@CoordinatedHeadState{seenTxs, seenUTxO}, previousState}, NetworkEvent (ReqTx _ tx)) ->
     case applyTransactions ledger seenUTxO [tx] of
       Left (_, err) -> Wait $ WaitOnNotApplicableTx err -- The transaction may not be applicable yet.
       Right utxo' ->
         let newSeenTxs = seenTxs <> [tx]
          in nextState
-              ( OpenState parameters $
-                  headState
-                    { seenTxs = newSeenTxs
-                    , seenUTxO = utxo'
-                    }
+              ( OpenState
+                  { parameters
+                  , coordinatedHeadState =
+                      headState
+                        { seenTxs = newSeenTxs
+                        , seenUTxO = utxo'
+                        }
+                  , previousState
```

Comment:
So the `previousState` is only updated on transitions? This would not be clear to me from the name and I would highly appreciate a haddock string in the definition of `HeadState` explaining what `previousState` is.